### PR TITLE
LPD-33901 Use role name in workflow xml definition

### PR DESCRIPTION
**Pull request by igor-franca**
Related [bug](https://liferay.atlassian.net/browse/LPD-33901) - Release blocker

After a [change](https://github.com/brianchandotcom/liferay-portal/pull/153052/files#diff-97219d8219276d8f05d1a198731ff6fd5d3230ef7c99e858c65d8a5c7d7b342eR77) in the role API return, the FE of the process builder, which previously used the [role key disguised as an external reference code](https://github.com/liferay/liferay-portal/blob/e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3/modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/shared-components/BaseRoleType.js#L105), started adding role externalReferenceCode values ​​in the workflow definition XML, where the role name values ​​should be added.

As a proposed solution, I modified the role API to return the name value in the name field instead of the title, since the name value was no longer being returned.

Now we just use the term role name in the process builder FE, since the role key is just an abstraction of the role name.

For the `RoleNameContainingMultibyteCharactersCanBeUsedInWorkflow` poshi test to be corrected and continue to make sense, some changes needed to be made.

Changes not related to the process builder will be reviewed in the commerce repository

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-test/src/testFunctional/tests/ProcessBuilderKaleoDesignerReact.testcase [hunk 1]
```diff
@@ -4845,7 +4845,7 @@ definition {
 		property test.liferay.virtual.instance = "false";
 		property test.run.type = "single";
 
-		task ("Given a regular role with multi-byte characters in the title") {
+		task ("Given a regular role with multi-byte characters in the name") {
```
Comment (cgmonte) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-test/src/testFunctional/tests/ProcessBuilderKaleoDesignerReact.testcase:4848]: Please, check if this is done through the UI or API . If UI, then this should be "key".

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi, these changes make a lot of sense, specially getting rid of `displayName` and `roleKey` in the front-end code. it simplifies and deletes properties that does not exist in API responses. Let's wait for the review on the back-end side, as it's a breaking change and may not be accepted. My only request for change is the title of a test. Please, check my comment. Otherwise, LGTM.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-test/src/testFunctional/tests/ProcessBuilderKaleoDesignerReact.testcase (hunk 1 above)
Comment (igor-franca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-test/src/testFunctional/tests/ProcessBuilderKaleoDesignerReact.testcase:4848]: Hello Carlos, we are sending the roleKey in the name property of the API within JSONRole.addRegularRole. 

What do you think?

**Comment by igor-franca** [comment]
ci:test:workflow

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
ci:report:69079665

**Comment by igor-franca** [comment]
resent [here](https://github.com/brianchandotcom/liferay-portal/pull/153529)

**Pull request closed without merging**